internal/pkg/gateway/registry.go

```diff
@@ -139,6 +140,9 @@ layout:
 		}
 	}
 
+	// shuffle the layouts - keep the preferred ones first
+	rand.Shuffle(len(preferredLayouts), func(i, j int) { preferredLayouts[i], preferredLayouts[j] = preferredLayouts[j], preferredLayouts[i] })
```

Comment:
Do you have to call rand.Seed() first?
https://yourbasic.org/golang/shuffle-slice-array/

Reply:
Ok, confirmed it is not needed.